modal/runner.py

```diff
@@ -502,6 +503,9 @@ async def _deploy_app(
     if client is None:
         client = await _Client.from_env()
 
+    if not client.authenticated:
+        raise ExecutionError("Objects cannot be created with an unauthenticated client")
```

Comment:
The copy-paste made me wonder if we should have a `client.check_authenticated()` or something so the error types/ messages don't drift over time, but probably not worth it.